graphql_compiler/compiler/emit_cypher.py

```diff
@@ -1,8 +1,31 @@
 # Copyright 2019-present Kensho Technologies, LLC.
 """Convert lowered IR basic blocks to Cypher query strings."""
+from graphql_compiler.compiler.helpers import Location
 from .blocks import Fold, QueryRoot, Recurse, Traverse
 
 
+def _get_full_path_location_name(fold_scope_location):
+    """Return a unique name with the full traversal path to this FoldScopeLocation."""
+    # HACK(Leon): Get a unique name for each vertex in a fold traversal in Cypher.
+    # For FoldScopeLocation objects, get_location_name() only uses the first edge in the traversal,
+    # which doesn't work for Cypher because we need to explicitly label every intermediate vertex
+    # along that path.
+    # For other query languages like MATCH and Gremlin, the fold directive doesn't require
+    # us to name all the intermediate vertices on the path, which is why this is Cypher-specific.
+    base_location = fold_scope_location.base_location
+    base_query_path = base_location.query_path  # the path traversed so far
+    fold_path = fold_scope_location.fold_path  # the path specified at or within the folded scope.
+    full_path = base_query_path + tuple(u'_'.join(edge_name) for edge_name in fold_path)
+    location = Location(full_path, field=base_location.field,
+                        visit_counter=base_location.visit_counter)
+    if base_location.field is not None:
+        raise ValueError(u'Expected base_location\'s field to be None since this method is used to'
```

Comment:
`to` and `traverse` on the next line don't have a space between them, so when Python concatenates the pieces together, you'll get `totraverse` as one word. Same as `afield` below.